### PR TITLE
[WORKFLOW] fix publish artifact #1508

### PR DESCRIPTION
**Pull request by mmd-afegbua**
Introduces nullglob to expand artifact pattern search(*) to null if empty.

`create-pr-artifact` does not create any package if PUBLISH_PR_ARTIFACT != 1.
`handler.publish-pr-packages` skips all other steps if there is no package from the workflow that triggered it.

### CODE REVIEW

**Comment by hellwolf** [comment]
@mmd-afegbua 

1. please setup signed commits.
2. keep PR in draft mode if not finished yet, just in case.

**Comment by hellwolf** [comment]
image

**Comment by mmd-afegbua** [comment]
Set to ready to test the Create AR Artifact pipeline.

File: .github/workflows/handler.publish-pr-packages.yml [hunk 1]
```diff
@@ -61,6 +61,7 @@ jobs:
           pr_number="${{ steps.pr.outputs.pr_number }}"
           tasks/npmrc-use-github.sh > pr-packages/$pr_number/.npmrc # using GITHUB_TOKEN
           cd pr-packages/$pr_number
+          shopt -s nullglob
```
Comment (hellwolf) [inline .github/workflows/handler.publish-pr-packages.yml:69]: okay

File: .github/workflows/handler.publish-pr-packages.yml [hunk 2]
```diff
@@ -49,8 +49,8 @@ jobs:
         run: |
           ls -R pr-packages
           pr_number=$(ls pr-packages)
-          # find num of files and print a dot for each file and count the dots
-          n_packages=$(find "pr-packages/$pr_number" -maxdepth 1 -type f -printf '.' | wc -c)
+          # Get the number of packages in the artifact
+          n_packages=$(find "pr-packages/$pr_number" -type f | wc -l | xargs)
```
Comment (hellwolf) [inline .github/workflows/handler.publish-pr-packages.yml]: this doesn't make sense to me

**Comment by mmd-afegbua** [comment]
`handler.publish-pr-packages.yml` only runs in default branch. PR has to be merged for changes to take effect.

**Review by hellwolf — APPROVED** [review]

**Pull request merged**